### PR TITLE
add unit tests for render-helm-chart function and run unit tests in CI

### PR DESCRIPTION
**Pull request by natasha41575**
- `make unit-test` to run all the unit tests in CI
- adds several unit tests for the render-helm-chart function

/cc @mengqiy @jeremyrickard @KnVerey

### CODE REVIEW

File: krm-functions/scripts/unit-test.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+err=0
+for d in */ ; do
+  if [ $d != "scripts/" ]; then
+    for e in "$d"*/ ; do (
+      echo running unit tests for "$e" ...
+      cd "$e" || exit
+      go test -cover -v ./... || exit
```
Comment (mengqiy) [inline krm-functions/scripts/unit-test.sh:21]: Is there a problem if we run the go tests in the parent directory instead of `cd` to each subdirectory?

Comment (natasha41575) [reply krm-functions/scripts/unit-test.sh:21]: I get the following error:

```
go: cannot find main module, but found .git/config in {PATH}/sigs.k8s.io/krm-functions-registry
        to create a module there, run:
        cd .. && go mod init
```

Comment (mengqiy) [reply krm-functions/scripts/unit-test.sh:21]: Got it

File: krm-functions/scripts/unit-test.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+err=0
+for d in */ ; do
+  if [ $d != "scripts/" ]; then
+    for e in "$d"*/ ; do (
+      echo running unit tests for "$e" ...
+      cd "$e" || exit
```
[the remark this answers is not in the document]
Comment (natasha41575) [reply krm-functions/scripts/unit-test.sh]: Thank you! Did not know about `set -e`. I have fixed it.

**Review by mengqiy — APPROVED** [review]
/lgtm
/approve

**Pull request merged**